Engine/ComponentSettings.cs

```diff
@@ -627,17 +627,19 @@ public virtual void Initialize()
         /// <summary>
         /// Sets current settings for a component setting based on a stream of the file contents of a ComponentSettings XML file.
         /// </summary>
+        /// <exception cref="InvalidDataException">If the input stream is not valid XML</exception>
         /// <param name="xmlFileStream">The component settings stream to be set</param>
         /// <returns></returns>
-        public static void SetCurrent(Stream xmlFileStream) =>
-            context.SetCurrent(xmlFileStream);
+        public static void SetCurrent(Stream xmlFileStream) => context.SetCurrent(xmlFileStream);
 
         /// <summary>
         /// Sets current settings for a component setting based on a stream of the file contents of a ComponentSettings XML file.
         /// </summary>
-        /// <param name="xmlFileStream"></param>
+        /// <exception cref="InvalidDataException">If the input stream is not valid XML</exception>
+        /// <param name="xmlFileStream">The component settings stream to be set</param>
+        /// <param name="errors">Any XML errors that occurred during deserialization</param>
         /// <returns></returns>
-        public static XmlError[] SetFromXmlStream(Stream xmlFileStream) => context.SetCurrent(xmlFileStream);
+        public static void SetCurrent(Stream xmlFileStream, out XmlError[] errors) => errors = context.SetCurrent(xmlFileStream);
```

Comment:
Should we switch to IEnumerable<XmlError> instead?

Also, how complicated would it be to have this overload never throw?

Reply:
It is easy to return an XmlError instead of throwing `InvalidDataException`, but it seems weird to wrap a generic exception in an XmlError